Week1/assets/HYFRepo.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const gitHub = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+
+function createAppendElement(htmlTag, parent) {
+
+    const node = document.createElement(htmlTag);
+    parent.appendChild(node);
+    return node;
+}
+
+function content() {
+
+    const main = document.getElementById('root');
+
+    const header = createAppendElement('header', main);
+    header.setAttribute('class', 'selection-wrapper');
+
+    const label = createAppendElement('label', header);
+    label.setAttribute('class', 'labelcls');
+    label.textContent = 'Select Repository' + ' ';
+
+    const toSelect = createAppendElement('select', header);
+    toSelect.setAttribute('id', 'selections');
+
+    const wrapperDiv = createAppendElement('div', main);
+    wrapperDiv.setAttribute('class', 'main-wrapper');
+
+    const repoInfoDiv = createAppendElement('div', wrapperDiv);
+    repoInfoDiv.setAttribute('id', 'repo-Info');
+
+    const createUlIno = createAppendElement('ul', repoInfoDiv);
+    createUlIno.setAttribute('id', 'repo-info-ul');
+
+    const contInfoDiv = createAppendElement('div', wrapperDiv);
+    contInfoDiv.setAttribute('id', 'cont-info');
+
+    const contSection = createAppendElement('p', contInfoDiv);
+    contSection.setAttribute('class', 'section-name');
+    contSection.textContent = 'Contributions';
+
+    const ulContInfo = createAppendElement('ul', contInfoDiv);
+    ulContInfo.setAttribute('id', 'cont-info-ul');
+
+
+    retrieveData(gitHub, (error, data) => {
+        if (error !== null) {
+            console.error(error.message);
+        } else {
+            constructOptions(data);
+        }
+    });
+}
+
+
+function retrieveData(url, callBack) {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true);
+    xhr.responseType = "json";
+    xhr.onreadystatechange = () => {
+        const repositories = xhr.response;
+        if (xhr.readyState === 4) {
+            if (xhr.status < 400) {
+                callBack(null, repositories);
+            } else {
+                callBack(new Error(xhr.statusText));
+            }
+        }
+    };
+
+    xhr.send();
+}
+
+function constructOptions(repos) {
+    const select = document.getElementById('selections');
+    repos.forEach((repo, i) => {
+        const options = createAppendElement('option', select);
+        options.innerHTML = repos[i].name;
+        options.setAttribute('value', i);
+
```

Comment:
The options in the drop-down menu are not sorted alphabetically

Reply:
Also this one I covered in week 2 assignment.